### PR TITLE
(DOC-3947) Update auth.conf docs

### PR DESCRIPTION
**Pull request by clairecadman**
This commit removes EOL documentation and links to more detailed examples.

### CODE REVIEW

File: documentation/config_file_auth.markdown [hunk 1]
```diff
@@ -142,7 +126,7 @@ A `match-request` can take the following parameters, some of which are required:
 
     > **Note:** While the new HOCON format does not provide a direct equivalent to the [deprecated][] `method` parameter's `search` indirector, you can create the equivalent rule by passing GET and POST to `method` and specifying endpoint paths using the `path` parameter.
 
--   **`query-params` and `environment`:** For endpoints on a Puppet 4 master, you can supply the `environment` as a query parameter suffix on the request's base URL. Use the optional `query-params` setting and provide the list of query parameters as an array to the setting's `environment` parameter.
+-   **`query-params` and `environment`:** Use the optional `query-params` setting and provide the list of query parameters as an array to the setting's `environment` parameter.
```
Comment (Magisus) [inline documentation/config_file_auth.markdown]: Do we know if "environment" is part of tk-auth here (as in, that's the only value that can be in here), or if it's just being referenced as the most common example of a query param for our APIs?

I'll look into this.

Comment (Magisus) [reply documentation/config_file_auth.markdown]: It looks like "environment" is not actually part of tk-auth, but just an example. I think we should reword this to more generally describe query-params. Something like:

-   **`query-params`:** Use the optional `query-params` setting and provide the list of query parameters. Each entry is a hash of the param name followed by a list of its values.

And then the example we already have below is good.

Comment (clairecadman) [reply documentation/config_file_auth.markdown]: Thanks @Magisus! I just updated this section. Let me know if there is anything else or if you're happy for it to be merged.
